### PR TITLE
Clean up color and smiley handling

### PR DESCRIPTION
**Pull request by kflynn**
Best reviewed commit by commit. This PR tries to rationalize the way we handle colors and smileys, as well as cleaning up some README and chart commentary.

- **Rationalize color handling: funnel through Color.lookup, and default to blue**
   - Note that the colors are now Paul Tol's "Bright" scheme, but I use "blue" for the color he calls "cyan", and "darkblue" for the one he calls "blue". This is because the darker blue gets into more trouble with colorblindness.
- **Update logo-128.png with new color**
   - This has the correct border and is using the correct blue background.
- **Rationalize smiley handling, too, funnelling through Smileys.Lookup**
- **Switch "smiling" to "grinning" because that's the actual name of the smiley.**
- **Clean up chart to allow color and smiley to not unnecessarily duplicate setting defaults**
- **Clean up README and chart description**
- **Minor GUI tweaks**
   - These are things like making the cell border thicker, improving the display of the pod info when it's turned on, and correctly interpolating the user-agent name under the hood.

### CODE REVIEW

File: pkg/faces/constants.go [hunk 1]
```diff
@@ -17,42 +17,102 @@
...
+		"white":     "white",
+
+		// These are colorblindness-friendly values from the "Tol" palette, as
+		// noted above.
+		"violet":    "#332288",
+		"green":     "#117733",
+		"bluegreen": "#44AA99",	// best avoided, see below
+		"blue":      "#88CCEE",
+		"yellow":    "#DDCC77",
+		"coral":     "#CC6677",	// best avoided, see below
```
Comment (mikemorris) [inline pkg/faces/constants.go:70]: IIUC the `Defaults` below intends to reference this color for timeouts and rate limits but refers to `"pink"` still instead.

File: pkg/faces/constants.go [hunk 2]
```diff
@@ -17,42 +17,102 @@
...
-	"smiley": Smileys["Cursing"],
+	"color":  "grey",
+	"smiley": "Cursing",
 
 	// 504 errors (GatewayTimeout) from the face workload will get handled in
 	// the GUI, but from the color & smiley workloads, they should get
 	// translated to a pink color and a sleeping face.
-	"color-504":  Colors["pink"],
-	"smiley-504": Smileys["Sleeping"],
+	"color-504":  "pink",
```
Comment (mikemorris) [inline pkg/faces/constants.go:112]:
```suggestion
	"color-504":  "coral",
```

File: pkg/faces/constants.go [hunk 3]
```diff
@@ -17,42 +17,102 @@
...
+	"smiley-504": "Sleeping",
 
 	// Ratelimits are pink with an exploding head.
-	"color-ratelimit":  Colors["pink"],
-	"smiley-ratelimit": Smileys["Kaboom"],
+	"color-ratelimit":  "pink",
```
Comment (mikemorris) [inline pkg/faces/constants.go:116]:
```suggestion
	"color-ratelimit":  "coral",
```

File: pkg/faces/constants.go [hunk 4]
```diff
@@ -17,42 +17,102 @@
...
+		"red":       "#882255", // be careful, see below
+
+		// Bluegreen and coral should be avoided, full stop, since they are very
+		// similar to each other for folks with protanopia or deuteranopia.
+		//
+		// The red/green pair and the green/blue pair can be similar enough for
+		// folks with some forms of colorblindness that it might be a bad idea to
+		// rely on just the color for semantics: e.g. if red means "bad" and green
+		// means "good", it might not be immediately apparently what's up if the
+		// display shows all red or all green. Red/cyan and green/violet are much
```
Comment (mikemorris) [inline pkg/faces/constants.go:81]: Unclear what color `cyan` is referring to - is it `blue` or `bluegreen`? Should fix these names for consistency.

File: assets/html/index.html [hunk 5]
```diff
@@ -26,21 +26,37 @@
...
+		.bg-green {
+			background-color: #117733;		/* Tol palette: green */
+		}
+
+		.bg-red {
+			background-color: #882255; 	/* Tol palette: red/magenta */
+		}
+
+		.bg-blue {
+			background-color: #88CCEE; 	/* Tol palette: cyan */
```
Comment (mikemorris) [inline assets/html/index.html:53]:
```suggestion
			background-color: #88CCEE; 	/* Tol palette: light blue */
```
`cyan` is unclear if it means light blue or the `bluegreen` color

File: assets/html/index.html [hunk 6]
```diff
@@ -26,21 +26,37 @@
...
 		}
 
 		.red {
-			color: rgb(125 42 83); /* dark magenta */
+			color: #882255; 	/* Tol palette: red/magenta */
 		}
 
 		.blue {
-			color: rgb(151 202 234); /* light blue */
+			color: #88CCEE; 	/* Tol palette: cyan */
```
Comment (mikemorris) [inline assets/html/index.html:37]:
```suggestion
			color: #88CCEE; 	/* Tol palette: light blue */
```
`cyan` is unclear if it means light blue or the `bluegreen` color

File: pkg/faces/constants.go [hunk 7]
```diff
@@ -17,42 +17,102 @@
...
-var Colors = map[string]string{
-	"grey":    "grey",
-	"purple":  "rgb(48 34 130)",
-	"green":   "rgb(55 117 59)",
-	"cyan":    "rgb(55 117 59)",   // too similar to pink and grey, avoid
-	"blue":    "rgb(151 202 234)", // light blue
-	"yellow":  "rgb(218 204 130)",
-	"pink":    "rgb(191 108 120)",
-	"hotpink": "rgb(158 75 149)",
-	"red":     "rgb(125 42 83)", // dark magenta
```
Comment (mikemorris) [inline pkg/faces/constants.go:41]: Check for places elsewhere where some of these older names were used, specifically:
- [ ] `purple` -> `violet`
- [ ] `cyan` -> `bluegreen`
- [ ] `pink` -> `coral`
- [ ] `hotpink` -> `fuschia`

**Review by mikemorris — CHANGES_REQUESTED** [review]
With the color renaming I think there's a handful of spots elsewhere that need to be updated to consistently use the new names.

File: README.md [hunk 8]
```diff
@@ -53,6 +57,33 @@ In here you will find:
 - To run the demo as we've given it before, check out [DEMO.md]. The easiest
   way to use that is to run it with [demosh].
 
+## Architecture
+
+The Faces architecture is fairly simple:
+
+- The `faces-gui` workload, reached on the `/faces/` path, just returns the
```
[the remark this answers is not in the document]
Comment (kflynn) [reply README.md:64]: We only did that in the Gateway API workshop, yeah. So far I haven't changed it here.

File: assets/logo-128.png
[the remark this answers is not in the document]
Comment (kflynn) [reply assets/logo-128.png:1]: It was, yes. 🤦‍♂️ Great catch!

File: pkg/faces/constants.go (hunk 1 above)
Comment (kflynn) [reply pkg/faces/constants.go]: I'm resolving all these because you were right. 🤦‍♂️ Fixed now!

**Comment by kflynn** [comment]
Thanks for the catch on the half-done color cleanup – this has the rest of the color change (🤦‍♂️) and more cleanup.

**Review by mikemorris — APPROVED** [review]
Ran through the workshop script at https://github.com/BuoyantIO/gateway-api-workshop/ with these changes and everything looked good!

**Pull request merged**